### PR TITLE
External CI: Switch all pipeline GPU_TARGETS to gfx942

### PR DESCRIPTION
**Pull request by alexxu-amd**
The test phase of external CI will be conducted on a MI300 system. Thus, switch all GPU_TARGETS to gfx942. Successful builds can be found: https://dev.azure.com/ROCm-CI/ROCm-CI/_build?definitionScope=%5Cexperimental%5Cgfx942-test

### CODE REVIEW

File: .azuredevops/components/rocWMMA.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
         -DCMAKE_BUILD_TYPE=Release
         -DROCWMMA_BUILD_TESTS=ON
         -DROCWMMA_BUILD_SAMPLES=OFF
-        -DGPU_TARGETS=gfx1100
+        -DAMDGPU_TARGETS=gfx942
```
Comment (amd-jmacaran) [inline .azuredevops/components/rocWMMA.yml:68]: does this work with the different flag name?

Comment (alexxu-amd) [reply .azuredevops/components/rocWMMA.yml:68]: Their github readme uses AMDGPU_TARGETS: https://github.com/ROCm/rocWMMA
It was either a recent change or we have been using the wrong name for setting gpu targets

Comment (danielsu-amd) [reply .azuredevops/components/rocWMMA.yml:68]: The Cmakelists.txt does use AMDGPU_TARGETS, line 111: https://github.com/ROCm/rocWMMA/blob/develop/CMakeLists.txt#L111

**Review by amd-jmacaran — APPROVED** [review]

File: .azuredevops/components/AMDMIGraphX.yml [hunk 2]
```diff
@@ -87,7 +87,7 @@ jobs:
         -DCMAKE_CXX_COMPILER=$(Agent.BuildDirectory)/rocm/llvm/bin/clang++
         -DCMAKE_C_COMPILER=$(Agent.BuildDirectory)/rocm/llvm/bin/clang
         -DCMAKE_BUILD_TYPE=Release
-        -DAMDGPU_TARGETS=gfx1030;gfx1100
+        -DAMDGPU_TARGETS=gfx942
```
Comment (danielsu-amd) [inline .azuredevops/components/AMDMIGraphX.yml:90]: MIGraphX looks to use GPU_TARGETS:
https://github.com/ROCm/AMDMIGraphX/blob/develop/Jenkinsfile#L133

**Comment by danielsu-amd** [comment]
hipBLAS and hipSOLVER look like they don't need a target flag to be set at all:
https://github.com/ROCm/hipBLAS/pull/335
https://github.com/ROCm/hipSOLVER/pull/13

hipSPARSELt needs its flag to be changed to gfx942:
https://github.com/ROCm/ROCm/blob/amd/alexxu12/testMI300Target/.azuredevops/components/hipSPARSELt.yml#L78

File: .azuredevops/components/AMDMIGraphX.yml (hunk 2 above)
Comment (alexxu-amd) [reply .azuredevops/components/AMDMIGraphX.yml:90]: Aha, good catch. Let me retest using the correct name

Comment (danielsu-amd) [reply .azuredevops/components/AMDMIGraphX.yml]: Build looks good: https://dev.azure.com/ROCm-CI/ROCm-CI/_build/results?buildId=3963&view=logs&j=b0b0eb5e-9de9-5eb1-7212-04e102eae4f9

**Review by danielsu-amd — APPROVED** [review]
Looks good 👍

**Pull request merged**